lnrpc/routerrpc/config.go

```diff
@@ -3,26 +3,25 @@ package routerrpc
 import (
 	"time"
 
-	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/btcsuite/btcutil"
 )
 
 // RoutingConfig contains the configurable parameters that control routing.
 type RoutingConfig struct {
-	// PenaltyHalfLife defines after how much time a penalized node or
-	// channel is back at 50% probability.
-	PenaltyHalfLife time.Duration
-
-	// PaymentAttemptPenalty is the virtual cost in path finding weight
-	// units of executing a payment attempt that fails. It is used to trade
-	// off potentially better routes against their probability of
-	// succeeding.
-	PaymentAttemptPenalty lnwire.MilliSatoshi
-
-	// MinProbability defines the minimum success probability of the
-	// returned route.
-	MinRouteProbability float64
+	// MinRouteProbability is the minimum required route success probability
+	// to attempt the payment.
+	MinRouteProbability float64 `long:"minrtprob" description:"Minimum required route success probability to attempt the payment"`
 
 	// AprioriHopProbability is the assumed success probability of a hop in
 	// a route when no other information is available.
-	AprioriHopProbability float64
+	AprioriHopProbability float64 `long:"apriorihopprob" description:"Assumed success probability of a hop in a route when no other information is available."`
+
+	// PenaltyHalfLife defines after how much time a penalized node or
+	// channel is back at 50% probability.
+	PenaltyHalfLife time.Duration `long:"penaltyhalflife" description:"Defines the duration after which a penalized node or channel is back at 50% probability"`
+
+	// AttemptCost is the virtual cost in path finding weight units of
+	// executing a payment attempt that fails. It is used to trade off
+	// potentially better routes against their probability of succeeding.
+	AttemptCost btcutil.Amount `long:"attemptcost" description:"The (virtual) cost in sats of a failed payment attempt"`
```

Comment:
what's the rationale for changing this to sats, when we later convert it back to msats? this will make anyone's current config values be off by a factor of 1000 after updating

Reply:
It was already in sats. The diff may be a bit confusing, because the command line options struct and the internal config struct are unified in this pr.

Reply:
ah i see, thanks for explaining